fullscreen.bs

```diff
@@ -199,29 +199,63 @@ these steps:
+   <!-- cross-process; check is only needed on pending as it is recursive already -->
 
    <li><p><a>Fullscreen is supported</a>.
 
    <li><p>This algorithm is <a>allowed to request fullscreen</a>.
   </ul>
 
+ <li><p>Let <var>pendingDoc</var> be <var>pending</var>'s <a>node document</a>.
+
+ <li><p>Let <var>topLevelDoc</var> be <var>pendingDoc</var>'s <a>top-level browsing context</a>'s
+ <a>active document</a>. <!-- cross-process -->
+
+ <li><p>Let <var>resize</var> be true if <var>error</var> is false and <var>topLevelDoc</var>'s
+ <a>fullscreen element</a> is null, and false otherwise.
+ <!-- cross-process -->
+
  <li><p>Return <var>promise</var>, and run the remaining steps <a>in parallel</a>.
 
- <li><p>If <var>error</var> is false: Resize <var>pending</var>'s
- <a>top-level browsing context</a>'s <a>active document</a>'s viewport's dimensions to match the
- dimensions of the screen of the output device. Optionally display a message how the end user can
- revert this.
+ <li><p>If <var>resize</var> is true, resize <var>topLevelDoc</var>'s viewport to match the
+ dimensions of the screen. Optionally display a message how the end user can revert this.
  <!-- cross-process -->
 
  <li>
   <p>As part of the next <a>animation frame task</a>, run these substeps:
 
   <ol>
-   <li><p>If either <var>error</var> is true or the <a>fullscreen element ready check</a> for
-   <var>pending</var> returns false, <a>fire an event</a> named <code>fullscreenerror</code> on
-   <var>pending</var>'s <a>node document</a>, reject <var>promise</var> with a
-   <code>TypeError</code> exception, and terminate these steps.
-   <!-- cross-process; check is only needed on pending as it is recursive already -->
+
+   <li>
+    <p>If any of the following conditions are false, then set <var>error</var> to true:
+
+    <ul>
+     <li><p><var>pending</var>'s <a>node document</a> is <var>pendingDoc</var>.
+
+     <li><p>The <a>fullscreen element ready check</a> for <var>pending</var> returns true.
+     <!-- cross-process; check is only needed on pending as it is recursive already -->
+
+     <li><p><var>topLevelDoc</var>'s viewport matches the dimensions of the screen.
+     <!-- cross-process -->
+    </ul>
+
+    <p class=note>For example, <var>pending</var> may have been moved to another document,
+    <code>allowfullscreen</code> attributes may have been removed, or <var>topLevelDoc</var> may
+    have exited fullscreen since {{Element/requestFullscreen()}} was invoked.
+
+   <li>
+    <p>If <var>error</var> is true, run these subsubsteps:
+
+    <ol>
+     <li><p><a>Fire an event</a> named <code>fullscreenerror</code> on <var>pendingDoc</var>.
+
+     <li><p>Reject <var>promise</var> with a <code>TypeError</code> exception.
+
+     <li><p>Terminate these steps and run the remaining subsubsteps <a>in parallel</a>.
+
+     <li><p>If <var>resize</var> is true, resize <var>topLevelDoc</var>'s viewport to its "normal"
```

Comment:
It would be a case like this:
```JavaScript
button.addEventListener('click', () => {
  a.requestFullscreen();
  b.requestFullscreen()
  b.remove();
});
```

Both requests will try to resize, and per spec it's a racy situation, but let's assume that after the resize the animation frame task for *a* runs first and *a* becomes the fullscreen element. Then *b* will end up in this situation, exiting fullscreen, but *a* will still be the fullscreen element.